src/interfaces.jl

```diff
@@ -398,12 +487,45 @@ eval(default_plot_signatures(:plot, :plot!, :Any))
 
 # plots to scene
 
+plotfunc(::Combined{F}) where F = F
+
+
+
 """
 Main plotting signatures that plot/plot! route to if no Plot Type is given
 """
-function plot!(scene::SceneLike, ::Type{Any}, attributes::Attributes, args...)
-    PlotType = plottype(to_value.(args)...)
-    plot!(scene, PlotType, attributes, args...)
+function plot!(scene::SceneLike, P::PlotFunc, attributes::Attributes, args...; kw_attributes...)
+    attributes = merge!(Attributes(kw_attributes), attributes)
+    argvalues = to_value.(args)
+    PreType = plottype(P, argvalues...)
+    # plottype will lose the argument types, so we just extract the plot func
+    # type and recreate the type with the argument type
+    PreType = Combined{plotfunc(PreType), typeof(argvalues)}
+    keys = used_attributes(PreType, argvalues...)
+    kw_signal = if isempty(keys) # lift(f) isn't supported so we need to catch the empty case
+        Node(())
+    else
+        lift((args...)-> Pair.(keys, args), getindex.(attributes, keys)...) # make them one tuple to easier pass through
+    end
+    # call convert_arguments for a first time to get things started
+    converted = convert_arguments(PreType, argvalues...; kw_signal[]...)
+    # convert_arguments can return different things depending on the recipe type
+    # apply_conversion deals with that!
+    FinalType, argsconverted = apply_convert!(PreType, attributes, converted)
+    converted_node = Node(argsconverted)
+    input_nodes =  to_node.(args)
+    onany(kw_signal, lift(tuple, input_nodes...)) do kwargs, args
```

Comment:
Somehow I feel here it'd be good to add some "Trait" to these functions (meaning, the "listeners"). I think that there will be several occasions where the user doesn't want to call `convert_arguments` again when just one argument changes. Say for example that I'm calling it on `x,y` that have to be of equal length. If I make `x` longer I'd like to wait that I've also made `y` shorter. So I would add a "is_convert_argument" trait to these updater so that the user can do:

```julia
setindex!(x, x_new, notify = !is_convert_argument)
setindex!(y, y_new)
```